### PR TITLE
meta: turn this repo into a monorepo

### PR DESCRIPTION
**Pull request by ephys**
## Pull Request Checklist

- [ ] Have you added new tests to prevent regressions?
- [ ] If a documentation update is necessary, have you opened a PR to [the documentation repository](https://github.com/sequelize/website/)? 
- [ ] Did you update the typescript typings accordingly (if applicable)?
- [ ] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Does the name of your PR follow [our conventions](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md#6-commit-your-modifications)?

## Description Of Change

Writing plugins & packages that depend on core is a headache without a proper monorepo setup. This PR sets that up.

This should make it trivial to create new packages such as [`@sequelize/postgres`](https://github.com/sequelize/sequelize/issues/13722), and [`@sequelize/validator.js`](https://github.com/sequelize/sequelize/issues/15497)

I'm sure I have missed something. Let's find out 🤞

I haven't tested the publish setup. If it fails, I'll open a follow-up PR to debug it

### CODE REVIEW

**Comment by WikiRik** [comment]
Let's publish a new alpha on the latest channel after we merge this to check if it works.

Another thing we can't really check right now and might need to fix in a follow up PR is the behaviour of renovate

File: packages/validator-js/package.json
Comment (WikiRik) [inline packages/validator-js/package.json:19]:
```suggestion
```

File: packages/validator-js/package.json
Comment (WikiRik) [inline packages/validator-js/package.json:2]: Will this be released as well? Or does it start with only `@sequelize/core`?

File: packages/validator-js/package.json
Comment (WikiRik) [reply packages/validator-js/package.json:19]: There are some things we should add as well but I don't think they're for this PR.

File: packages/core/package.json
Comment (WikiRik) [inline packages/core/package.json:195]:
```suggestion
    "---------------------------------------- integration tests --------------------------------------------": "",
```
Or however long we need it to be to the same size as the others

File: .github/workflows/ci.yml
Comment (WikiRik) [inline .github/workflows/ci.yml:39]: Which dependencies are in `./node_modules`?

**Review by WikiRik — COMMENTED** [review]
Haven't really worked with `yarn workspaces` before and might need to get used to it a bit, especially with running the scripts using `yarn workspace @sequelize/core test-unit-all` for example. In a future PR we might want to look into making some scripts available on the root package as well (`test-unit-all` and `test-integration-{dialect}` comes to my mind first)

File: .github/workflows/ci.yml
Comment (WikiRik) [reply .github/workflows/ci.yml:39]: Considering that for the `typedoc` check we still install the root dependencies

File: packages/validator-js/package.json
Comment (ephys) [reply packages/validator-js/package.json:2]: The goal is to publish a dummy release, to see if it works

File: packages/validator-js/package.json
Comment (ephys) [reply packages/validator-js/package.json:19]: Yep, follow-up PR :)

File: .github/workflows/ci.yml
Comment (ephys) [reply .github/workflows/ci.yml:39]: Yarn workspace dedupes dependencies needed by the different packages of the workspace by putting them in the root node_modules

Got to remove the redundant install from the docs script

Comment (ephys) [reply .github/workflows/ci.yml:39]: Done

File: packages/core/package.json
Comment (ephys) [reply packages/core/package.json:195]: Done

**Review by WikiRik — APPROVED** [review]
I think we should be able to test this out

**Comment by ephys** [comment]
Had to merge the changes made by renovate. We'll need to merge this before renovate updates something again

Also changed how the CI installs node_modules: Running `yarn install` at the root of the repo installs the dependencies of all packages automatically. And removed the lockfile from `packages/core`, because yarn doesn't actually use it. It uses the one at the root of the repo

**Review by WikiRik — APPROVED** [review]

**Comment by ephys** [comment]
🙌

**Pull request merged**
